### ISSUE TITLE
Template Creation Fail - Error: unexpected end of JSON input

### ISSUE THREAD

**Issue by PhilipWhiteside** [CiscoDevNet/terraform-provider-mso#26]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

* v1.0.0

### MSO version

* V 3.3.1e
* Physical Nexus Dashboard: 2.0(2g)

### APIC version and APIC Platform for Site Level Resources

* V 4.2(6d) on-prem

### Affected Resource(s)

* mso_schema_template

### Terraform Configuration Files

```hcl
terraform {
  required_providers {
    mso = {
      source  = "CiscoDevNet/mso"
      version = "0.1.5"
    }
  }
}
locals {
  key = "pwhitesi"
  users = [
    {
      name             = "admin"
      id               = "48d105bdfbc49a5fcf39a10b961386e61dfe40085cb0315d818b761c5735dafa"
      security_domains = []
    }
  ]
  sites = [
    {
      name = "site_id_london"
      id   = "5e84e7990f0000d0002385a4"
    },
    {
      name = "site_id_manchester"
      id   = "5e84e7d80f0000eb002385a5"
    }
  ]
}

resource "mso_tenant" "primary" {
  name         = local.key
  display_name = local.key

  dynamic "user_associations" {
    for_each = local.users
    content {
      user_id = user_associations.value["id"]
    }
  }

  dynamic "site_associations" {
    for_each = local.sites
    content {
      site_id          = site_associations.value["id"]
      security_domains = try(site_associations.value["security_domains"], null)
    }
  }
}

resource "mso_schema" "primary" {
  name          = local.key
  template_name = "global"
  tenant_id     = mso_tenant.primary.id
}

resource "mso_schema_template" "london" {
  schema_id    = mso_schema.primary.id
  name         = "london"
  display_name = "london"
  tenant_id    = mso_tenant.primary.id
}
```

### Debug Output

https://gist.github.com/PhilipWhiteside/49a5319b13d9af2c2e08c67eb3946eeb

### Expected Behavior

Creation of the template and success in the Terraform output. 

### Actual Behavior

Template is created, but Terraform fails. 

![Screenshot from 2021-06-18 13-20-39 749222608](https://user-images.githubusercontent.com/22427598/122561849-3d4bf200-d03a-11eb-927a-686880ea574e.png)

```
│ Error: unexpected end of JSON input
│ 
│   with mso_schema_template.london,
│   on main.tf line 27, in resource "mso_schema_template" "london":
│   27: resource "mso_schema_template" "london" {
```
### Steps to Reproduce

1. `terraform apply`

### References

* #0000

**Comment by JoaoUrtigaMelo** [comment]
I`m getting the same error for different resources. Sometimes it was created and sometimes it doesn`t.

Error: unexpected end of JSON input
│ 
│   with mso_schema_template_vrf.vrf,
│   on main.tf line 53, in resource "mso_schema_template_vrf" "vrf":
│   53: resource "mso_schema_template_vrf" "vrf" {
│ 
╵
╷
│ Error: unexpected end of JSON input
│ 
│   with mso_schema_template_anp.anp,
│   on main.tf line 98, in resource "mso_schema_template_anp" "anp":
│   98: resource "mso_schema_template_anp" "anp" {
│ 
╵
╷
│ Error: unexpected end of JSON input
│ 
│   with mso_schema_template_filter_entry.any,
│   on main.tf line 129, in resource "mso_schema_template_filter_entry" "any":
│  129: resource "mso_schema_template_filter_entry" "any" {
│ 
╵
╷
│ Error: unexpected end of JSON input
│ 
│   with mso_schema_template_filter_entry.http,
│   on main.tf line 138, in resource "mso_schema_template_filter_entry" "http":
│  138: resource "mso_schema_template_filter_entry" "http" {
│ 
╵
╷
│ Error: unexpected end of JSON input
│ 
│   with mso_schema_template_filter_entry.ssh,
│   on main.tf line 164, in resource "mso_schema_template_filter_entry" "ssh":
│  164: resource "mso_schema_template_filter_entry" "ssh" {
│ 
╵
╷
│ Error: unexpected end of JSON input
│ 
│   with mso_schema_template_filter_entry.icmp,
│   on main.tf line 177, in resource "mso_schema_template_filter_entry" "icmp":
│  177: resource "mso_schema_template_filter_entry" "icmp" {
│ 
╵
╷
│ Error: unexpected end of JSON input
│ 
│   with mso_schema_template_filter_entry.mysql,
│   on main.tf line 187, in resource "mso_schema_template_filter_entry" "mysql":
│  187: resource "mso_schema_template_filter_entry" "mysql" {
│

**Comment by anvitha-jain** [comment]
This issue is resolved, and the changes are in #52. This will be soon be available in the next release.
